shared/protocol/src/main/java/io/pravega/shared/protocol/netty/WireCommands.java

```diff
@@ -1696,13 +1697,178 @@ public static WireCommand readFrom(ByteBufInputStream in, int length) throws IOE
         }
     }
 
+    @Data
+    public static final class GetTableKeys implements Request, WireCommand {
+
+        final WireCommandType type = WireCommandType.GET_TABLE_KEYS;
+        final long requestId;
+        final String segment;
+        final String delegationToken;
+        final int suggestedKeyCount;
+        final ByteBuffer continuationToken; // this used to indicate the point from which the next keys should be fetched.
+
+        @Override
+        public void process(RequestProcessor cp) {
+            cp.getTableKeys(this);
+        }
+
+        @Override
+        public void writeFields(DataOutput out) throws IOException {
+            out.writeLong(requestId);
+            out.writeUTF(segment);
+            out.writeUTF(delegationToken == null ? "" : delegationToken);
+            out.writeInt(suggestedKeyCount);
+            out.writeInt(continuationToken.remaining()); // continuation token length.
+            if (continuationToken.remaining() != 0) {
+                out.write(continuationToken.array(), continuationToken.arrayOffset() + continuationToken.position(), continuationToken.remaining());
```

Comment:
This presumes an array backed byte array. This already triggered a bug once, this should implement it the same way we did for event:
https://github.com/pravega/pravega/pull/2712/files#diff-599b4d30bd4eb53895a9c32146254bb0R396

Table key and table value should be similarly updated. I missed that in code review.

Reply:
done, updated the code for all ByteBuffer instances.